### PR TITLE
fix unit test of influxd in the windows

### PR DESCRIPTION
**Pull request by runner-mei**
fix unit test of influxd in the windows

### CODE REVIEW

File: cmd/influxd/backup/backup.go [hunk 1]
```diff
@@ -360,7 +360,10 @@ Options:
 func retentionAndShardFromPath(path string) (retention, shard string, err error) {
 	a := strings.Split(path, "/")
 	if len(a) != 3 {
-		return "", "", fmt.Errorf("expected database, retention policy, and shard id in path: %s", path)
+		a = strings.Split(path, "\\")
```
Comment (corylanou) [inline cmd/influxd/backup/backup.go]: I'm unfamiliar with what the contents of the path would be, but would `os.PathSeparator` be a better method and do something like:

```
a := srings.Split(path, os.PathSeparator)
```

If not, then we may want to make a specific windows/unix file for this specific function, as it feels the issue is platform specific.

**Comment by corylanou** [comment]
Looks good.  One question to be addressed. Thanks!

File: services/meta/config.go [hunk 2]
```diff
@@ -96,15 +96,11 @@ func (c *Config) Validate() error {
...
-	host, port, err := net.SplitHostPort(addr)
-	if err != nil {
-		return addr
-	}
-
-	if host == "" {
-		return net.JoinHostPort(DefaultHostname, port)
+	address, err := DefaultHost(DefaultHostname, addr)
+	if nil != err {
+		panic(err)
```
Comment (jwilder) [inline services/meta/config.go]: The previous behavior was to return `addr` here.  It should not panic.

**Comment by runner-mei** [comment]
all unit test will ok while this patch is merged.
I hope merge it as soon as possible. I have a holiday soon.

**Comment by jwilder** [comment]
Thanks @runner-mei!

**Pull request merged**